### PR TITLE
Upgrade to @guardian/pan-domain-node

### PR DESCRIPTION
**Pull request by mbarton**
Upgrade to allow us to sunset the old standalone pan-domain-node repo.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -2,99 +2,22 @@
...
-			}
-		},
-		callback: (err, policy) => {
-			test.equal(policy.policyDocument.Statement[0].Effect, 'Deny');
-			test.end();
-		}
-	});
-});
-
-tap.test('fails when the cookie is not set', function (test) {
```
Comment (mbarton) [inline test/index.js:48]: This is now completely delegated to the library so I've removed the test

**Comment by mbarton** [comment]
Currently failing when running with:

```
2020-04-19T20:30:14.147Z	a1a0e15d-4cbb-4bc6-8ea7-e4ad396c5ea4	ERROR	TypeError: Cannot read property 'AuthenticationStatus' of undefined
    at verifyUser (/var/task/index.js:468:32)
    at /var/task/index.js:551:24
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

File: package.json [hunk 2]
```diff
@@ -19,24 +19,21 @@
...
     "eslint": "^6.7.1",
     "exec-chainable": "0.0.3",
-    "nodemon": "^2.0.1",
     "node-riffraff-artefact": "^2.0.1",
-    "rollup": "^0.27.0",
-    "rollup-plugin-babel": "^2.5.0",
-    "rollup-plugin-commonjs": "^2.2.1",
-    "rollup-plugin-node-resolve": "^1.5.0",
+    "nodemon": "^2.0.1",
+    "rollup": "^2.6.1",
```
Comment (mbarton) [inline package.json:29]: Updated roll-up as `@guardian/pan-domain-node` currently has a circular import dependency which older versions couldn't handle:

We should fix the library but this lets us upgrade independently. The newer version emits a nice warning describing the error:

```
(!) Circular dependencies

node_modules/@guardian/pan-domain-node/dist/src/api.js -> node_modules/@guardian/pan-domain-node/dist/src/panda.js -> node_modules/@guardian/pan-domain-node/dist/src/api.js

node_modules/@guardian/pan-domain-node/dist/src/api.js -> node_modules/@guardian/pan-domain-node/dist/src/panda.js -> /home/travis/build/guardian/pan-domain-lambda/node_modules/@guardian/pan-domain-node/dist/src/api.js?commonjs-proxy -> node_modules/@guardian/pan-domain-node/dist/src/api.js
```

**Comment by mbarton** [comment]
Fixed the error (due to a circular dependency) by upgrading rollup.

Tested and working in `CODE`

File: src/environment.js [hunk 3]
```diff
@@ -3,4 +3,16 @@ const stage = (process.env.AWS_LAMBDA_FUNCTION_NAME || '')
...
 
-export const STAGE = stage || 'UNKNOWN';
+function getSettingsFile (stage) {
+	switch (stage) {
+		case 'PROD':
+			return 'gutools.co.uk.settings.public';
+		case 'CODE':
+			return 'code.dev-gutools.co.uk.settings.public';
+		default:
+			return 'local.dev-gutools.co.uk.settings.public';
```
Comment (akash1810) [inline src/environment.js:13]: So.
 
 
 
 
 
Much.
 
 
 
 
 
Whitespace!

File: src/index.js [hunk 4]
```diff
@@ -1,36 +1,38 @@
...
-	.then(key => validate(cookie, key))
-	.then(user => {
-		callback(null, policy(
-			`${user.firstName} ${user.lastName} <${user.email}>`,
-			'Allow',
-			events.methodArn
-		));
+	panda.verify(cookie)
+	.then(({ status, user }) => {
+		if (status === 'Authorised') {
```
Comment (akash1810) [inline src/index.js:14]: `status` doesn't match numerical HTTP Status codes? 😢

Comment (mbarton) [reply src/index.js:14]: Maybe not the best choice of variable names but it is at least a [Typescript enum](https://github.com/guardian/pan-domain-authentication/blob/master/pan-domain-node/src/api.ts#L3) in the API, the idea being you can then map that on to framework you're using. For example I don't think you can control the status code returned from the custom authorizer (maybe?), instead just returning `Allow` or `Deny` to the callback

**Pull request merged**